plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php

```diff
@@ -21,13 +21,6 @@ public function __construct( $task_list ) {
 
 		add_action( 'show_admin_bar', array( $this, 'possibly_hide_wp_admin_bar' ) );
 
-		// Use "switch_theme" instead of "after_switch_theme" because the latter is fired after the next WP load and we don't want to trigger action when switching theme to TT3 via onboarding theme API.
-		global $_GET;
-		$theme_switch_via_cys_ai_loader = isset( $_GET['theme_switch_via_cys_ai_loader'] ) ? 1 === absint( $_GET['theme_switch_via_cys_ai_loader'] ) : false;
-		if ( ! $theme_switch_via_cys_ai_loader ) {
-				add_action( 'switch_theme', array( $this, 'mark_task_as_complete' ) );
-		}
-
```

Comment:
> We had some failing tests and after some debugging we got to this lines and saw the task was being marked as complete when switching theme, which the test is doing in the beforeAll making the tests fail 

@alba I think we should retain these lines because it is intentional to mark the CYS task as completed when the user switches to another theme without passing the `theme_switch_via_cys_ai_loader` parameter.

Perhaps a better way to make the e2e pass is to reset the `woocommerce admin customize store completed` option in `beforeAll`.

Reply:
Thanks @chihsuan, I did that on https://github.com/woocommerce/woocommerce/pull/43736/commits/a587508ff81830ac7478e042a9b9d4b2a8b6d2e8 and worked 🙌